tests/expected/derive-invariant/safety_constraint_helper_funs/safety_constraint_helper_funs.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that functions can be called in the `#[safety_constraint(...)]` attribute helpers.
+//! This is like the `invariant_helper` test but using a function instead
+//! of passing in a predicate.
+
+extern crate kani;
+use kani::Invariant;
+
+#[derive(kani::Arbitrary)]
+#[derive(kani::Invariant)]
+struct PositivePoint {
+    #[safety_constraint(is_coordinate_safe(x))]
+    x: i32,
+    #[safety_constraint(is_coordinate_safe(y))]
+    y: i32,
+}
+
+fn is_coordinate_safe(val: &i32) -> bool {
+    *val >= 0
+}
+
+#[kani::proof]
+fn check_invariant_helper_ok() {
+    let pos_point: PositivePoint = kani::any();
+    assert!(pos_point.is_safe());
+}
+
+#[kani::proof]
+#[kani::should_panic]
+fn check_invariant_helper_fail() {
+    // In this case, `kani::any()` returns unconstrained values that don't
+    // respect the invariants.
```

Comment:
nit: This comment took me by surprise, but looking at the code I see what you mean. The case is that building the struct from arbitrary values is not the same as creating an arbitrary struct.

Reply:
I've rewritten this comment, hope it's clear now.